### PR TITLE
Deadleaves overhaul [RDY]

### PR DESCRIPTION
**Pull request by BorkBorkGoesTheCode**
This is the earlier overhaul mod+ #18817, without migration. Some improvements have been made.

Testing: I deleted the Deadleaves mod from one of my experimentals, downloaded this one plonked it into mod folder, started a game with it. No errors.

No migration was used here.

### CODE REVIEW

File: data/mods/FictonalWeapons/modinfo.json [hunk 1]
```diff
@@ -2,9 +2,9 @@
     "type": "MOD_INFO",
     "mod_type": "SUPPLEMENTAL",
     "ident": "FIC_Weapons",
-    "name": "DeadLeaves' Fictional Guns",
-    "description": "Adds a bunch of rare weapons loosely based fiction. ",
+    "name": "Fictional Guns Pack",
+    "description": "Adds a bunch of rare and exotic larger-than-life weapons.  Based on DeadLeaves' Fictional Guns.",
     "category": "items",
-    "author": "DeadLeaves",
+    "authors": [ "DeadLeaves", "Cataclysm-DDA community" ],
```
Comment (illi-kun) [inline data/mods/FictonalWeapons/modinfo.json]: Mention of community is useless here - entire game is made by community. Why you didn't put your name here?

**Comment by DangerNoodle** [comment]
I am unsure why the modified MP5 is being removed. That (and a 5.56x45mm revolver, which I am surprised was not added) would be a natural choice for the mod, being a notable movie and video game weapon.

File: data/mods/FictonalWeapons/modinfo.json (hunk 1 above)
Comment (BorkBorkGoesTheCode) [inline data/mods/FictonalWeapons/modinfo.json]: Because I'm not the only contributor to the mod.

**Comment by Reclusive-reptile** [comment]
There is already a 5.56x45 revolver in core. The L2037 Backup, found in military locations.

**Comment by DeadLeaves** [comment]
Hey, I'm a bit out of the loop, but what's going on here? 

From what I can tell the mod has been changed to a more universal mod for all kinds of fictional weapons instead of just the ones I made. That's a great change and I don't mind at all.

 It's also great that you went ahead and fixed a lot of issues in my absence. I really do appreciate it.

What I don't get is why you're changing my weapons fundamentally instead of adding new ones yourself?

**Comment by BorkBorkGoesTheCode** [comment]
>What I don't get is why you're changing my weapons fundamentally instead of adding new ones yourself?

I was worried that their presence in the game would put the project at risk of a lawsuit, so I changed them.

**Comment by DeadLeaves** [comment]
I see, that's a valid point. Though for the mp5 an under slung grenade launcher should be a non-issue since it's used in both movies and other video games. [half life](http://half-life.wikia.com/wiki/MP5) 

On the other hand, I totally agree on the rico gun, but if anything, a simple name change should suffice. at most a slight change in description.  Or make it use magazines instead of being a revolver. 

Don't get me wrong though, I really like the guns you've turned them into. It's just that I kinda liked the old ones as well, and don't see a reason why they can't co-exist.

**Comment by DangerNoodle** [comment]
As I stated when I found the first pull request of this sort, and cited the End of Days entry on IMFDB.

One question though, is this .44 Magnum SMG referencing a "real" fictional weapon, or was it made up?

**Comment by BorkBorkGoesTheCode** [comment]
>On the other hand, I totally agree on the rico gun, but if anything, a simple name change should suffice. at most a slight change in description. Or make it use magazines instead of being a revolver. 

Don't get me wrong though, I really like the guns you've turned them into. It's just that I kinda liked the old ones as well, and don't see a reason why they can't co-exist.

@DeadLeaves I agree. The MP5 EOD was the best 9mm smg in the game. Since the stats appear to be your creation, I will rewrite the descriptions and re-add them into the game.

>One question though, is this .44 Magnum SMG referencing a "real" fictional weapon, or was it made up?

Made up. I heard a discussion about the existence and feasibility of .357 and .44 magnum smgs on the net, and started thinking about what such a gun would look like.

**Comment by Rivet-the-Zombie** [comment]
>I was worried that their presence in the game would put the project at risk of a lawsuit

D'you really think that Universal Pictures is going to try to claim in court that they invented the Heckler and Koch Maschinenpistole 5 so that they can sue the developers of a free PC game?

**Comment by DangerNoodle** [comment]
Now if the concern was H&K in general it would almost make sense, but then they would have already sued over the inclusion of their weaponry in mainline.

**Comment by DeadLeaves** [comment]
When I made the MP5EOD, I copied the stats of the mainline mp5 and made it slightly worse (wear and tear from modifying it). And frankly I don't really care about stats and such on the weapons. I just really like the ideas of the weapons I made back then. And I LOVE the fact that you've fixed the issues they had. Go ahead by all means, change the stats to make them balanced with the mainline or whatever you feel is balanced. 

I'll repeat it, I love what you're doing and keep up the good work.

**Comment by BorkBorkGoesTheCode** [comment]
This should be ready.

**Comment by BorkBorkGoesTheCode** [comment]
@Rivet-the-Zombie @DeadLeaves do you have any objections?

**Comment by DeadLeaves** [comment]
I'll put it this way. I don't mind at all if it's left like this. However I wouldn't mind the revolver still having the launcher.  (Only the grenade launcher part, I don't mind the rename, in fact the new name sounds cool). But I also love the idea of a flamer pistol, I think we should have both. You could make them being from the same gunsmith with minor name suffix change. That'd be pretty cool.

The only nitpick would be the line "WARNING! DO NOT EAT!" It sounds a bit silly.

**Comment by DangerNoodle** [comment]
@DeadLeaves, there has additionally been discussion on https://github.com/CleverRaven/Cataclysm-DDA/pull/19037 regarding merging this mod with Icecoon's Weapons Pack. What is your opinion regarding that idea?

I am on the fence regarding that because your mod is primarily about "real" fictional weapons, whereas IWP is focused on the author's own creations. But as far as I can tell, the Eagle 1776 and other odd additions appear to be made up as well, which changes the intent of this mod.

**Comment by mugling** [comment]
Having multiple mods with broadly similar content slows down development significantly and puts a lot of burden on the core developers to keep them up to date.

To be clear there are no 'owners' of any code or mods. The issue here is purely whether or not it is good for the project as a whole to merge the various fictional gun mods.

We don't have infinite resources so reducing the workload is a necessary step. If anyone feels strongly enough about a mod to feel they 'own' it then it should be developed externally - otherwise what you are effectively asking for is your own way and our collective assistance in carrying that out.

We have a huge number of abandoned mods that are a good testament to how easy it is to create something then opine about it versus actively maintaining something.

**Comment by DangerNoodle** [comment]
Understandable. Their feedback is still useful though.

**Comment by BorkBorkGoesTheCode** [comment]
>I'll put it this way. I don't mind at all if it's left like this. However I wouldn't mind the revolver still having the launcher. (Only the grenade launcher part, I don't mind the rename, in fact the new name sounds cool). But I also love the idea of a flamer pistol, I think we should have both. You could make them being from the same gunsmith (ill suggest D&B or DB engraving, for deadleaves & borkbork) with minor name suffix change. That'd be pretty cool.

The grenade launcher was already restored in the "bugfixes and the readdition of guns" commit. The warning was an indirect reference to the NPC habit of eating lighters and BOOM.

**Comment by BorkBorkGoesTheCode** [comment]
error 500, for some reason

**Comment by BorkBorkGoesTheCode** [comment]
Wow, GitHub support works fast.

**Comment by BorkBorkGoesTheCode** [comment]
Thank you Coolthulhu.

**Pull request merged**